### DISCUSSION TITLE
How to output functor material properties

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#20307] [Q&A General] [answered]
Hello:

I use an AuxVariable and the ADFunctorElementalAux to get the value of a material property.  For example:
```
[rho_out]
    type = ADFunctorElementalAux
    functor = 'rho_gas'
    variable = 'rho_var'
    execute_on = 'timestep_end'
  []
```
However, both in a VPP and in the Exodus file, the values of rho_var correspond to that for the last element in my mesh.  Values for the other elements don't show up.  Is there some way to do this?

Thanks,
  Joe Kelly

**Comment by lindsayad** [comment, marked as answer]
@joe61vette can you show me your block for your `rho_var` aux variable? My suspicion is that you are creating a nodal aux variable, in which case you will be evaluating a nodal aux kernel ... and we use `_current_elem` in the `FunctorElementalAux` evaluation which will be wrong if the aux kernel is being evaluated as a nodal quantity. We will add an error check to `FunctorElementalAux` that will catch this in the future

**Reply by joe61vette** [reply]
I bet you're right.  Here it is:

  [rho_var]
  []

It was copied from the NS test case functorfluidprops.i

**Reply by GiudGiud** [reply]
Please change it to 
```
[rho_var]
  type = MooseVariableFVReal
[]
```

that test will be modified. The exodus seems fine because all the variables are constant over the domain in that test

**Comment by joe61vette** [comment]
THANKS Alex & Guillaume.
